modules/miner/miner.go

```diff
@@ -194,18 +210,21 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
+		return err
+	}
+
 	m.mu.Lock()
 	defer m.mu.Unlock()
```

Comment:
Is it OK that the `m.tg.stopFns` (`m.cs.Unsubscribe` and `m.tpool.Unsubscribe`) are called outside of the lock?